uportal-war/src/main/java/org/apereo/portal/layout/dlm/remoting/UpdatePreferencesServlet.java

```diff
@@ -231,6 +232,66 @@ public ModelAndView removeElement(HttpServletRequest request, HttpServletRespons
     }
 
     /**
+     * Remove the first element with the provided fname from the layout.
+     *
+     * @param request HttpServletRequest
+     * @param response HttpServletResponse
+     * @param fname fname of the portlet to remove from the layout
+     * @return json response
+     * @throws IOException if the person cannot be retrieved
+     */
+    @RequestMapping(method = RequestMethod.POST, params = "action=removeByFName")
+    public ModelAndView removeByFName(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            @RequestParam(value = "fname", required = true) String fname
+    ) throws IOException {
+        ModelAndView result = new ModelAndView("jsonView", Collections.EMPTY_MAP);
+        try {
+            IUserInstance ui = userInstanceManager.getUserInstance(request);
+            IPerson per = getPerson(ui, response);
+
+            UserPreferencesManager upm = (UserPreferencesManager) ui.getPreferencesManager();
+            IUserLayoutManager ulm = upm.getUserLayoutManager();
+
+            String elementId = ulm.getUserLayout().findNodeId(new PortletSubscribeIdResolver(fname));
+            if (elementId != null) {
```

Comment:
Could the logical structure be [simplified with a guard clause](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)?